src/System.Devices.Gpio/System.Devices.Gpio.Samples/Bme280.cs

```diff
@@ -0,0 +1,419 @@
+        private const byte BME280_DIG_P2_REG = 0x90;
+        private const byte BME280_DIG_P3_REG = 0x92;
+        private const byte BME280_DIG_P4_REG = 0x94;
+        private const byte BME280_DIG_P5_REG = 0x96;
+        private const byte BME280_DIG_P6_REG = 0x98;
+        private const byte BME280_DIG_P7_REG = 0x9A;
+        private const byte BME280_DIG_P8_REG = 0x9C;
+        private const byte BME280_DIG_P9_REG = 0x9E;
+
+        private const byte BME280_DIG_H1_REG = 0xA1;
+        private const byte BME280_DIG_H2_REG = 0xE1;
+        private const byte BME280_DIG_H3_REG = 0xE3;
+        private const byte BME280_DIG_H4_REG = 0xE4;
+        private const byte BME280_DIG_H5_REG = 0xE5;
+        private const byte BME280_DIG_H6_REG = 0xE7;
+
+        private const byte BME280_REGISTER_CHIPID = 0xD0;
+        private const byte BME280_REGISTER_VERSION = 0xD1;
+        private const byte BME280_REGISTER_SOFTRESET = 0xE0;
+        private const byte BME280_REGISTER_CAL26 = 0xE1;
+        private const byte BME280_REGISTER_CONTROLHUMID = 0xF2;
+        private const byte BME280_REGISTER_CONTROL = 0xF4;
+        private const byte BME280_REGISTER_CONFIG = 0xF5;
+        private const byte BME280_REGISTER_PRESSUREDATA = 0xF7;
+        private const byte BME280_REGISTER_TEMPDATA = 0xFA;
+        private const byte BME280_REGISTER_HUMIDDATA = 0xFD;
+
+        // Structure to hold the calibration data that is
+        // programmed into the sensor in the factory
+        // during manufacture
+        private struct Bme280_Calibration_Data
+        {
+            public ushort dig_T1;
+            public short dig_T2;
+            public short dig_T3;
+            public ushort dig_P1;
+            public short dig_P2;
+            public short dig_P3;
+            public short dig_P4;
+            public short dig_P5;
+            public short dig_P6;
+            public short dig_P7;
+            public short dig_P8;
+            public short dig_P9;
+            public byte dig_H1;
+            public short dig_H2;
+            public byte dig_H3;
+            public short dig_H4;
+            public short dig_H5;
+            public sbyte dig_H6;
+        };
+
+        private Bme280_Calibration_Data _cal_data;
+
+        private float _temperature;
+        private float _humidity;
+        private float _pressure;
+        private int _tFine;
+
+        private SpiConnectionSettings _spiSettings;
+        private SpiDevice _spiDevice;
+
+        private Pin _csPin;
+        private Pin _mosiPin;
+        private Pin _misoPin;
+        private Pin _sckPin;
+
+        /// <summary>
+        /// Use this for hardware SPI.
+        /// </summary>
+        public Bme280(Pin cs, SpiConnectionSettings spiSettings)
+        {
+            _csPin = cs;
+            _spiSettings = spiSettings;
+        }
+
+        /// <summary>
+        /// Use this for software SPI.
+        /// </summary>
+        public Bme280(Pin cs, Pin mosi, Pin miso, Pin sck)
```

Comment:
cs should be "chipSelectLine" 
sck should be "serialClock"
miso could be "masterInSlaveOut"
mosi could be "masterOutSlaveIn"

Can you think of better names for miso/mosi that are more approachable to people who don't know these concepts?

Reply:
Ok, but I think people who will use this API or sample must have to understand what are those names because they need to connect the wires between the device and sensor and they use those names.